request-manager/grapher/grapher.go

```diff
@@ -342,8 +342,29 @@ func (o *Grapher) buildComponent(name string, nodeDefs map[string]*NodeSpec, nod
 				}
 
 				// Insert all components between the start and end vertices.
+				// Place in parallel if possible, otherwise serialize.
+				// Each parallel component is wrapped between dummy nodes.
+				currG, err := o.newEmptyGraph("repeat_"+n.Name, nodeArgs)
+				if err != nil {
+					return nil, err
+				}
+				prev := g.First
+				var count uint = 0
 				for _, c := range componentsForThisNode {
-					g.insertComponentBetween(c, g.First, g.Last)
+					currG.insertComponentBetween(c, currG.First, currG.Last)
+					count++
+					if n.Parallel != nil && count == *n.Parallel {
```

Comment:
Rather than checking in every iteration if parallel != nil, how about doing that check just once outside the loop and storing the value in another variable which you compare to `count` here. You could handle the case where it _is_ nil by setting `parallel = len(componentsForThisNode)`.